stdlib/filename.ml

```diff
@@ -326,7 +326,10 @@ let remove_extension name =
 external open_desc: string -> open_flag list -> int -> int = "caml_sys_open"
 external close_desc: int -> unit = "caml_sys_close"
 
-let prng_key = Domain.DLS.new_key Random.State.make_self_init
+let prng_key =
+  Domain.DLS.new_key
+    ~split_from_parent:Random.State.split
+    Random.State.make_self_init
```

Comment:
I suggest keeping the original code, without the `split_from_parent`.  (Note that there's similar code in hashtbl.ml that doesn't use `split_from_parent` either).

Why no splitting here?
1- Because there is no expectation of determinism or reproducibility.  For security reasons, this PRNG must not be predictable!  And that's why it's self-initialized from non-reproducible system data.
2- Because with splitting, the first domain creation will force the initialization of the PRNG of the initial domain, even if the whole program doesn't use `Filename.temp_file` at all !  I'd rather keep the "pay as you go" approach of OCaml 4, where the cost of self-initializing this PRNG is paid only by actual uses of `Filename.temp_file`.

Reply:
Makes sense, I'll do the change with an explanatory comment. Thanks!

(Your point (1) is not so convincing: it's not clear that splitting from a single system-initialized state is less predictable than system-initializing several times, especially on systems without /dev/(u)random where our poor emulation is used. But I'm sold on (2): the current approach means that, as soon as `temp_file` is called once, all new domain creations will result in PRNG initialization or splitting, which is wasteful.)

Reply:
>  it's not clear that splitting from a single system-initialized state is less predictable than system-initializing several times,

I agree.  The point I was trying to make is this: proper splitting is necessary for the default per-domain PRNG in module Random, because we can start from a known seed and want reproducibility in this case.  If you always start from a fresh seed (and you want to do that in Filename for security reasons) there is no reproducibility and splitting doesn't buy you much.

Reply:
Re: /dev/urandom, it is supported on Linux, macOS, the three BSDs, and Windows has its own code path, so the fallback code in `caml_unix_random_seed` is never exercised in practice.

Reply:
I changed this as suggested.